File: src/lib/api/modules/treehole.ts
import { get, post } from "../../http/client";
import type { PaginatedResponse } from "../types";

export type { PaginatedResponse } from "../types";

export interface TreeholeMessage {
  id: string;
  content: string;
  authorName?: string;
  isAnonymous: boolean;
  likeCount: number;
  createdAt: string;
}

export const treeholeApi = {
  getMessages: (page = 1, limit = 20) =>
    get<PaginatedResponse<TreeholeMessage>>("/api/treehole/messages", {
      page,
      limit,
    }),

  createMessage: (data: { content: string; isAnonymous: boolean }) =>
    post<TreeholeMessage>("/api/treehole/messages", data),
};

File: backend/app/modules/treehole.py
"""匿名树洞模块路由"""

from datetime import datetime

from fastapi import APIRouter, Request

from app.core.pagination import paginate
from app.core.response import ok, paginated
from app.data.treehole import TREEHOLE_MESSAGES

router = APIRouter(prefix="/api/treehole", tags=["treehole"])


@router.get("/messages")
def get_messages(page: int = 1, page_size: int = 20):
    items, total = paginate(TREEHOLE_MESSAGES, page, page_size)
    return paginated(items, total)


@router.post("/messages")
async def create_message(request: Request):
    body = await request.json()
    msg = {
        "id": f"th-{len(TREEHOLE_MESSAGES) + 1}",
        "content": body.get("content", ""),
        "is_anonymous": body.get("is_anonymous", True),
        "like_count": 0,
        "created_at": datetime.now().isoformat(),
    }
    TREEHOLE_MESSAGES.insert(0, msg)
    return ok({"id": msg["id"]})
